drivers/adc/adc_npcx.c

```diff
@@ -303,12 +297,14 @@ static const struct adc_driver_api adc_npcx_driver_api = {
 
 static int adc_npcx_init(const struct device *dev);
 
-static const struct npcx_alt adc_alts[] = NPCX_DT_ALT_ITEMS_LIST(0);
+PINCTRL_DT_INST_DEFINE(0);
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
+	"only one 'nuvoton_npcx_adc' compatible node may be present");
 
 static const struct adc_npcx_config adc_npcx_cfg_0 = {
 	.base = DT_INST_REG_ADDR(0),
 	.clk_cfg = NPCX_DT_CLK_CFG_ITEM(0),
-	.alts_list = adc_alts,
+	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(0),
```

Comment:
It's a programming error if the user fails to set the `pinmux` property on `&adc0_default`, correct?

Can you add a build assert to verify the user has configured at least one ADC channel?

Reply:
Done.